### ISSUE TITLE
pip-compile --generate-hashes fails on sphinx

### ISSUE THREAD

**Issue by aaugustin** [jazzband/pip-tools#438]
`pip-compile --generate-hashes` crashes on requirements files that contain `sphinx`.

##### Steps to replicate

```
myk@mYk:~ $ python -m venv test-pip-tools

myk@mYk:~ $ pip install pip-tools
Collecting pip-tools
  Using cached pip_tools-1.8.0-py2.py3-none-any.whl
Collecting first (from pip-tools)
  Using cached first-2.0.1-py2.py3-none-any.whl
Collecting click>=6 (from pip-tools)
  Using cached click-6.7-py2.py3-none-any.whl
Requirement already satisfied: six in ./.pyenv/versions/3.6.0/lib/python3.6/site-packages (from pip-tools)
Installing collected packages: first, click, pip-tools
Successfully installed click-6.7 first-2.0.1 pip-tools-1.8.0

myk@mYk:~ $ echo sphinx > requirements.in

myk@mYk:~ $ pip-compile --generate-hashes
Traceback (most recent call last):
  File "/Users/myk/.pyenv/versions/3.6.0/bin/pip-compile", line 11, in <module>
    sys.exit(cli())
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/piptools/scripts/compile.py", line 187, in cli
    hashes = resolver.resolve_hashes(results)
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/piptools/resolver.py", line 80, in resolve_hashes
    return {ireq: self.repository.get_hashes(ireq) for ireq in ireqs}
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/piptools/resolver.py", line 80, in <dictcomp>
    return {ireq: self.repository.get_hashes(ireq) for ireq in ireqs}
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/piptools/repositories/pypi.py", line 159, in get_hashes
    for candidate in matching_candidates
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/piptools/repositories/pypi.py", line 159, in <setcomp>
    for candidate in matching_candidates
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/piptools/repositories/pypi.py", line 166, in _get_file_hash
    download_dir=tmpdir, only_download=True, session=self.session
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/download.py", line 821, in unpack_url
    hashes=hashes
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/download.py", line 663, in unpack_http_url
    unpack_file(from_path, location, content_type, link)
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/utils/__init__.py", line 605, in unpack_file
    untar_file(filename, location)
  File "/Users/myk/.pyenv/versions/3.6.0/lib/python3.6/site-packages/pip/utils/__init__.py", line 577, in untar_file
    with open(path, 'wb') as destfp:
IsADirectoryError: [Errno 21] Is a directory: '/var/folders/pq/hzv7wgqs5fq0hf1bwzy4mlzr0000gn/T/tmpukek_v4nbuild/EXAMPLES'
```

##### Expected result

Should generate requirements.txt.

##### Actual result

Crashes with the trackback shown above.

**Comment by di** [comment]
This also happens with `pytest`:

```
NotADirectoryError: [Errno 20] Not a directory: '/var/folders/px/dlfyqxyx3mx0t_yx3dvbb9nc0000gn/T/tmp5ib4g7ldbuild/changelog/_template.rst'
```

**Comment by vphilippon** [comment]
Fixed by #557
